ds3-sdk/src/main/java/com/spectralogic/ds3client/metadata/MetaDataAccessImpl.java

```diff
@@ -0,0 +1,156 @@
+        permissions = permissions.replaceAll("w", "2");
+        permissions = permissions.replaceAll("x", "1");
+        permissions = permissions.replaceAll("-", "0");
+        final String ownerPerm = String.valueOf(Integer.parseInt(String.valueOf(permissions.charAt(0))) + Integer.parseInt(String.valueOf(permissions.charAt(1))) + Integer.parseInt(String.valueOf(permissions.charAt(2))));
+        final String groupPerm = String.valueOf(Integer.parseInt(String.valueOf(permissions.charAt(3))) + Integer.parseInt(String.valueOf(permissions.charAt(4))) + Integer.parseInt(String.valueOf(permissions.charAt(5))));
+        final String otherPerm = String.valueOf(Integer.parseInt(String.valueOf(permissions.charAt(6))) + Integer.parseInt(String.valueOf(permissions.charAt(7))) + Integer.parseInt(String.valueOf(permissions.charAt(8))));
+        final String totalPerm = ownerPerm + groupPerm + otherPerm;
+        return totalPerm + "(" + permString + ")";
+    }
+
+    //if os is windows then posix will not be called and we need to find permission in different manner
+    private void saveWindowsfilePermissions(final Path file, final Map<String, String> metadata) {
+        try {
+            final AclFileAttributeView view = Files.getFileAttributeView(file, AclFileAttributeView.class);
+            final List<AclEntry> acl = view.getAcl();
+            Set<AclEntryPermission> aclEntryPermissions;
+            String userType = "";
+            StringBuilder permission = null;
+
+            for (final AclEntry ac : acl) {
+                permission = new StringBuilder();
+                int index = 1;
+                aclEntryPermissions = ac.permissions();
+
+                for (final AclEntryPermission aclEntryPermission : aclEntryPermissions) {
+                    if (index == aclEntryPermissions.size())
+                        permission.append(aclEntryPermission.ordinal());
+                    else
+                        permission.append(aclEntryPermission.ordinal() + "_");
+                    index++;
+                }
+
+                userType = ac.principal().getName();
+
+                switch (userType) {
+                    case "BUILTIN\\Administrators":
+                        metadata.put("x-amz-meta-ds3-ownerPerm", permission.toString());
+                        break;
+
+                    case "NT AUTHORITY\\SYSTEM":
+                        metadata.put("x-amz-meta-ds3-authenticatedUser", permission.toString());
+                        break;
+
+                    case "NT AUTHORITY\\Authenticated Users":
+                        metadata.put("x-amz-meta-ds3-systemUser", permission.toString());
+                        break;
+
+                    case "BUILTIN\\Users":
+                        metadata.put("x-amz-meta-ds3-allUser", permission.toString());
+                        break;
+                }
+            }
+        } catch (final Exception e) {
+            e.printStackTrace();
```

Comment:
Exceptions should not be handled in this way.  At a minimum it should be logged with a slf4j logger, or wrapped in a runtime exception and re-thrown.